modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkExtensionModelParser.java

```diff
@@ -77,19 +77,17 @@ public List<ModelProperty> getAdditionalModelProperties() {
 
   @Override
   public String getName() {
-    return extensionName;
+    return metadataParser.getName();
```

Comment:
this feels unnecesary complex. Why introducing this concept of metadataParser (which by the way is a very confusing name since things like `name` are not metadata) when you can simply apply the template pattern to the MuleSdkModelParser?

Reply:
Again, this was just me having picked composition over inheritance. See [this other answer](https://github.com/mulesoft/mule/pull/11776#discussion_r988374631) for more details.
I'll try using the template pattern and see how it looks.

Reply:
This has been changed to use templating. It looks simpler now.